privacy/enforcer.go

```diff
@@ -1,43 +1,241 @@
 package privacy
 
-// PolicyEnforcer determines if personally identifiable information (PII) should be removed or anonymized per the policy.
-type PolicyEnforcer interface {
-	// CanEnforce returns true when policy information is specifically provided by the publisher.
-	CanEnforce() bool
+import (
+	"fmt"
+	"github.com/prebid/prebid-server/config"
+	"strings"
+)
 
-	// ShouldEnforce returns true when the OpenRTB request should have personally identifiable
-	// information (PII) removed or anonymized per the policy.
-	ShouldEnforce(bidder string) bool
+type ActivityResult int
+
+const (
+	ActivityAbstain ActivityResult = iota
+	ActivityAllow
+	ActivityDeny
+)
+
+type ActivityControl struct {
+	plans map[Activity]ActivityPlan
+}
+
+func NewActivityControl(privacyConf *config.AccountPrivacy) (ActivityControl, error) {
+	ac := ActivityControl{plans: nil}
```

Comment:
If Activities are not configured, we need to return ActivityControl to avoid null pointer exception in future usage:
```go
auctionReq.Activities.Allow(privacy.ActivityFetchBids, ...)
```
Plans should be set to nil, so it will handle undefined plan properly in `Allow`:
```go
plan, planDefined := e.plans[activity]

if !planDefined {
	return ActivityAbstain
}
```